supertokens/recipeModule.go

```diff
@@ -18,13 +18,14 @@ package supertokens
 import "net/http"
 
 type RecipeModule struct {
-	recipeID              string
-	appInfo               NormalisedAppinfo
-	HandleAPIRequest      func(ID string, req *http.Request, res http.ResponseWriter, theirHandler http.HandlerFunc, path NormalisedURLPath, method string) error
-	GetAllCORSHeaders     func() []string
-	GetAPIsHandled        func() ([]APIHandled, error)
-	HandleError           func(err error, req *http.Request, res http.ResponseWriter) (bool, error)
-	OnSuperTokensAPIError func(err error, req *http.Request, res http.ResponseWriter)
+	recipeID                   string
+	appInfo                    NormalisedAppinfo
+	HandleAPIRequest           func(ID string, req *http.Request, res http.ResponseWriter, theirHandler http.HandlerFunc, path NormalisedURLPath, method string) error
+	GetAllCORSHeaders          func() []string
+	GetAPIsHandled             func() ([]APIHandled, error)
+	GetAPIIdIfCanHandleRequest func(path NormalisedURLPath, method string) (*string, error)
```

Comment:
You should not change this struct type.. instead in the dashboard recipe, provide an implementation of `ReturnAPIIdIfCanHandleRequest` which overrides the one in the base class (this one)

Reply:
Updated as per our discussion.